### PR TITLE
mongodoc: introduce development revisions and development ACLs.

### PR DESCRIPTION
**Pull request by frankban**
Also implement the corresponding migration.

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1225/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1227/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1228/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1229/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1230/
Test PASSed.

File: internal/charmstore/migrations.go [hunk 1]
```diff
@@ -138,6 +142,36 @@ func addSupportedSeries(db StoreDatabase) error {
 	return nil
 }
 
+// handleDevelopmentEntities adds the Development field to all entities and
```
Comment (mhilton) [inline internal/charmstore/migrations.go:145]: handle doesn't seem to be the correct verb here perhaps it should be two migrations, addDevelopment & addDevelopmentACLs

File: internal/charmstore/migrations.go [hunk 2]
```diff
@@ -138,6 +142,36 @@ func addSupportedSeries(db StoreDatabase) error {
...
+// sets up ACLs for development revisions.
+func handleDevelopmentEntities(db StoreDatabase) error {
+	logger.Infof("adding development field to all entities")
+	if _, err := db.Entities().UpdateAll(nil, bson.D{{
```
Comment (mhilton) [inline internal/charmstore/migrations.go:149]: This should probably only update those that don't have a development field already. That way it's idempotent and its fairly easy to do.

**Comment by frankban** [comment]
Real pull request (against feature branch) can be found at https://github.com/juju/charmstore/pull/456

**Pull request closed without merging**